bindings/rust/evmc-declare/src/lib.rs

```diff
@@ -363,7 +363,15 @@ fn build_execute_fn(names: &VMNameSet) -> proc_macro2::TokenStream {
                 ::evmc_vm::EvmcContainer::<#type_name_ident>::from_ffi_pointer(instance)
             };
 
-            let result = container.execute(code_ref, &execution_context);
+            let result = ::std::panic::catch_unwind(|| {
```

Comment:
>It is not recommended to use this function for a general try/catch mechanism. The Result type is more appropriate to use for functions that can fail on a regular basis. Additionally, this function is not guaranteed to catch all panics, see the "Notes" section below.

https://doc.rust-lang.org/std/panic/fn.catch_unwind.html

Reply:
The main reason for this change is exactly listed in your link:
> It is currently undefined behavior to unwind from Rust code into foreign code, so this function is particularly useful when Rust is called from another language (normally C). This can run arbitrary Rust code, capturing a panic and allowing a graceful handling of the error.